ctapipe/image/morphology.py

```diff
@@ -110,28 +110,69 @@ def number_of_island_sizes(island_labels):
     return np.count_nonzero(small), n_medium, np.count_nonzero(large)
 
 
-def largest_island(islands_labels):
+def largest_island(island_labels):
     """Find the biggest island and filter it from the image.
 
     This function takes a list of islands in an image and isolates the largest one
     for later parametrization.
 
     Parameters
     ----------
-    islands_labels : array
+    island_labels : array
         Flattened array containing a list of labelled islands from a cleaned image.
         Second value returned by the function 'number_of_islands'.
 
     Returns
     -------
-    islands_labels : array
+    mask : array
         A boolean mask created from the input labels and filtered for the largest island.
         If no islands survived the cleaning the array is all False.
 
     """
-    if np.count_nonzero(islands_labels) == 0:
-        return np.zeros(islands_labels.shape, dtype="bool")
-    return islands_labels == np.argmax(np.bincount(islands_labels[islands_labels > 0]))
+    if np.count_nonzero(island_labels) == 0:
+        return np.zeros(island_labels.shape, dtype="bool")
+    return island_labels == np.argmax(np.bincount(island_labels[island_labels > 0]))
+
+
+def brightest_island(n_islands, island_labels, image):
+    """Find the brightest island and filter it from the image.
+
+    This function takes a list of islands in an image and the image itself
+    and isolates brightest island for later parametrization.
+
+    Parameters
+    ----------
+    n_islands: int
+        Total number of islands, first return value of `number_of_islands`
+    island_labels : array
+        Flattened array containing a list of labelled islands from a cleaned image.
+        Second value returned by the function 'number_of_islands'.
+
+    image: array
+        The image array
+
+    Returns
+    -------
+    mask : array
+        A boolean mask created from the input labels and filtered for the brightest island.
+        If no islands survived the cleaning the array is all False.
+    """
+
+    if n_islands == 0:
```

Comment:
you can use `island_labels` also for this check

Reply:
But checking the integer is cheaper than recomputing it using `np.count_nonzero`. Since this information is available, why not use it?